src/gpuhunt/providers/oci.py

```diff
@@ -0,0 +1,228 @@
+import logging
+import re
+from typing import Iterable, List, Optional, Type
+
+import oci
+from oci.identity.models import Region
+from pydantic import BaseModel, Field
+from requests import Session
+from typing_extensions import Annotated, TypedDict
+
+from gpuhunt._internal.constraints import KNOWN_GPUS
+from gpuhunt._internal.models import QueryFilter, RawCatalogItem
+from gpuhunt._internal.utils import to_camel_case
+from gpuhunt.providers import AbstractProvider
+
+logger = logging.getLogger(__name__)
+COST_ESTIMATOR_URL_TEMPLATE = "https://www.oracle.com/a/ocom/docs/cloudestimator2/data/{resource}"
+COST_ESTIMATOR_REQUEST_TIMEOUT = 10
+
+
+class OCICredentials(TypedDict):
+    user: Optional[str]
+    key_content: Optional[str]
+    fingerprint: Optional[str]
+    tenancy: Optional[str]
+    region: Optional[str]
+
+
+class OCIProvider(AbstractProvider):
+    NAME = "oci"
+
+    def __init__(self, credentials: OCICredentials):
+        self.api_client = oci.identity.IdentityClient(
+            credentials if all(credentials.values()) else oci.config.from_file()
+        )
+        self.cost_estimator = CostEstimator()
+
+    def get(
+        self, query_filter: Optional[QueryFilter] = None, balance_resources: bool = True
+    ) -> List[RawCatalogItem]:
+        shapes = self.cost_estimator.get_shapes()
+        products = self.cost_estimator.get_products()
+        regions: List[Region] = self.api_client.list_regions().data
+
+        result = []
+
+        for shape in shapes.items:
+            if (
+                shape.hidden
+                or shape.status != "ACTIVE"
+                or shape.shape_type.value != "vm"
+                or shape.sub_type.value == "flexible"
+            ):
+                continue
+
+            # extra validation, failing here would mean we are not handling some
+            # case that was not present in the data at the time of writing
+            if (
+                len(shape.products) != 1
+                or (ocpu_product := shape.products[0]).type.value != "ocpu"
+                or (product_details := products.find(ocpu_product.part_number)) is None
+                or product_details.billing_model != "UCM"
+                or product_details.price_type != "HOUR"
+                or (price_l10n := product_details.find_price_l10n("USD")) is None
+                or len(price_l10n.prices) != 1
+                or (product_price := price_l10n.prices[0]).model != "PAY_AS_YOU_GO"
+            ):
+                logger.warning(
```

Comment:
We usually don't write to the log about skipped offers.
```suggestion
                logger.debug(
```

Reply:
Skipping offers here would mean we are not handling cost estimator data properly. I think these messages should be visible so that we can notice them and adjust our code accordingly.

See the comment a few lines above.